airflow/models/__init__.py

```diff
@@ -2973,6 +2987,8 @@ class DagModel(Base):
     description = Column(Text)
     # Default view of the inside the webserver
     default_view = Column(String(25))
+    # Schedule interval
+    schedule_interval = Column(PickleType(pickler=dill))
```

Comment:
Do we need to use pickle here? Maybe better to use a  standardise string format? Although this is pretty hard also ;)

Reply:
Pickling was the simplest option, since we currently support cron-like strings, `datetime.timedelta`, and `dateutil.relativedelta`, but we could write our own serialization format instead. One simple option would be to store the type, along with whitelisted attributes, as json: `{"type": "relativedelta", "args": {"months": 2}}`. There are other fields we might need to add to `DagModel` that will also take some thinking about serialization, like `default_args`, and we might wind up pickling those too.

What do you think?

Reply:
I think I would like the json format. I did found some comments in the code that pickle will not be used at all anymore in 2.0. Although I'm not sure what the pickle roadmap exactly is inside airflow ;).

Reply:
I'm not sure about the pickle roadmap either, but dropping pickle for 2.0 would be great. I added an `Interval` type that knows how to serialize and deserialize `timedelta` and `relativedelta`. WDYT?